Source/NSDockTile.m

```diff
@@ -109,4 +132,74 @@ - (void) display
     }
 }
 
+- (void)draw
+{
+  NSMutableDictionary *attrs;
+  NSPoint text_location;
+  NSRect disc_rect;
+  NSSize disc_size;
+  int image_width, pad;
+  NSBezierPath *p;
+  NSPoint point;
+  CGFloat radius;
+  NSSize imageSize;
+
+  if (_showsApplicationBadge)
+    {
+      [_appIconImage compositeToPoint:NSZeroPoint operation:NSCompositeCopy];
+
+      imageSize = [_appIconImage size];
+      attrs = [[NSMutableDictionary alloc] init];
+      [attrs setObject: [NSFont boldSystemFontOfSize: imageSize.width/5]  forKey: NSFontAttributeName];
+      [attrs setObject: [NSColor blackColor]  forKey: NSForegroundColorAttributeName];
+
+      disc_size = [_badgeLabel sizeWithAttributes: attrs];
+
+      image_width = imageSize.width;
```

Comment:
This should be moved up a bit and used in line 153. Also a different name would be great as we use this value both for width and height and it is always confusing to see width and height in one computation.

Reply:
I cleaned up all width/height, removed some local variables too and use diff of sizes instead of fixed padding. How do you like it,  @fredkiefer ?